lib/heuristics/concerns/scheduling_end_phase.rb

```diff
@@ -30,20 +30,20 @@ def refine_solution(&block)
       add_missing_visits
     end
 
-    unless @services_data.all?{ |_id, data| data[:exclusion_cost].zero? }
+    unless @services_data.all?{ |_id, data| (data[:raw].exclusion_cost || 0).zero? }
```

Comment:
Default exclusion_cost is nil. I did not want to change it to zero because I assumed there would be an impact on ortools/dicho side. @braktar is there no impact ?

Reply:
With "What would be the consequence?" I was asking you if you can check there is an impact or not in fact..  Anyways, yes there is unfortunately :/ apparently tsp_simple expects "-1" for "no exclusion cost" so we cannot replace it with 0

```ruby
exclusion_cost: service.exclusion_cost && service.exclusion_cost.to_i || -1,
```

So I mark this resolved, this fix would need its own separate PR